Tasks/Common/codecoverage-tools/jacoco/jacoco.ant.ccenabler.ts

```diff
@@ -0,0 +1,179 @@
+/// <reference path="../../../../definitions/Q.d.ts" />
+/// <reference path="../../../../definitions/string.d.ts" />
+/// <reference path="../../../../definitions/vsts-task-lib.d.ts" />
+/// <reference path="../../../../definitions/node.d.ts" />
+
+import * as util from "../utilities";
+import * as tl from "vsts-task-lib/task";
+import * as ccc from "../codecoverageconstants";
+import * as cc from "../codecoverageenabler";
+import * as str from "string";
+import * as path from "path";
+import * as os from "os";
+import * as Q from "q";
+
+export class JacocoAntCodeCoverageEnabler extends cc.JacocoCodeCoverageEnabler {
+
+    reportDir: string;
+    excludeFilter: string;
+    includeFilter: string;
+    sourceDirs: string;
+    classDirs: string;
+
+    // -----------------------------------------------------
+    // Enable code coverage for Jacoco Ant Builds
+    // - enableCodeCoverage: CodeCoverageProperties  - ccProps
+    // -----------------------------------------------------    
+    public enableCodeCoverage(ccProps: { [name: string]: string }): Q.Promise<boolean> {
+        let _this = this;
+        let defer = Q.defer<boolean>();
+        _this.buildFile = ccProps["buildfile"];
+        _this.sourceDirs = ccProps["sourcedirectories"];
+        _this.classDirs = ccProps["classfilesdirectories"];
+        _this.reportDir = ccProps["reportdirectory"];
+
+        let classFilter = ccProps["classfilter"];
+        let filter = _this.extractFilters(classFilter);
+        _this.excludeFilter = _this.applyFilterPattern(filter.excludeFilter).join(",");
+        _this.includeFilter = _this.applyFilterPattern(filter.includeFilter).join(",");
+
+        return util.readXmlFileAsJson(_this.buildFile).
+            then(function (resp) {
+                _this.addCodeCoverageData(resp);
+            })
+            .thenResolve(true);
+    }
+
+    protected applyFilterPattern(filter: string): string[] {
+        let ccfilter = [];
+        let _this = this;
+
+        if (!util.isNullOrWhitespace(filter)) {
+            str(util.trimToEmptyString(filter)).replaceAll(".", "/").s.split(":").forEach(exFilter => {
```

Comment:
this is not about "path". this is pure cc tools pattern irrespective of platform